call-notes/call_6.md

```diff
@@ -0,0 +1,160 @@
+
+**Wen Xu:** How much time do I have? I am probably a little bit slow.
+
+**Jacob Creech:** Yeah, we have 11 minutes for both finishing questions but we can see how far we can get.
+
+**Wen Xu:** Okay, that's fine. I put all the information, all the details in the SIMD and I put more details in this slide. So, you are welcome to read the slide.
+
+#### The Silent Repair Phase
+
+Video |  [16:40](https://youtu.be/GA5AVg_svj8?t=1000)
+-|-
+
+**Wen Xu:** So, I am just going to generally introduce the silent repair phase and then we can go proceed to questions. So first of all when you restart a validator with this new arguments. Immediately we send the LastVotedForkSlots which is what you voted last and all the slots on this fork. And after that, everyone aggregates the LastVotedForkSlots from all the other restarted validators. And you could start repairing your slot if you think this slot could potentially have been up, it's confirmed before the restart. And we could draw a line somewhere to say these are the candidates which could have been confirmed before the slot and the other blocks I don't care. So, you repair all the blocks you care and after we repaired all of them, you aggregate all the last votes in the LastVotedForkSlots and choose your HeaviestFork.
+
+So, I think we are at 20 minutes now. I don't know whether I have introduced the new method enough so everyone has good grabs. But, we could we could see if anyone has any questions at this point. Hello, or I can proceed if no one has questions.
+
+**Jacob Creech:** Does anybody have any questions currently on the current approach or should we just continue?
+
+Alright go ahead and continue then `Wen`.
+
+#### Exit Silent Repair Phase
+
+Video |  [18:26](https://youtu.be/GA5AVg_svj8?t=1106)
+-|-
+
+**Wen Xu:** Okay! So, to exit a silent repair phase. I think one of the most important thing in outage handling is to make sure everyone is on the same page. Otherwise, you think everyone is on the same page, you single-handedly enter the restart and start making new blocks while everyone still repairing blocks, that would be disastrous. And a lot another I audit might be happening. So here we are very careful when we exit the silent repair phase. We would count whether enough people are ready for action. So the current check is whether 80% of the validators receipt the heaviest fork response from 80 percent of people. So we count 80% here because this is the current line we draw when we do a restart. We wait for 80% to join the restart, then we proceed. And also, even if we see that 80% of people respond, got response from 80%, we also linger for maybe we haven't totally decide the numbers here, we would say linger for two minutes because a gossip message propagation takes time. So linger for two minutes so that my HeaviestFork which contains how many responses I got can reach everyone.
```

Comment:
```suggestion
**Wen Xu:** Okay! So, to exit a silent repair phase. I think one of the most important thing in outage handling is to make sure everyone is on the same page. Otherwise, you think everyone is on the same page, you single-handedly enter the restart and start making new blocks while everyone still repairing blocks, that would be disastrous. And another outage might be happening. So here we are very careful when we exit the silent repair phase. We would count whether enough people are ready for action. So the current check is whether 80% of the validators received the heaviest fork response from 80% of people. So we count 80% here because this is the current line we draw when we do a restart. We wait for 80% to join the restart, then we proceed. And also, even if we see that 80% of people respond, got response from 80%, we also linger for maybe we haven't totally decide the numbers here, we would say linger for two minutes because a gossip message propagation takes time. So linger for two minutes so that my HeaviestFork which contains how many responses I got can reach everyone.
```